### ISSUE TITLE
Audit rules not uninstalled when osquery exits

### ISSUE THREAD

**Issue by Smjert** [osquery/osquery#7205]
# Bug report

### What operating system and version are you using?

Ubuntu 20.04

### What version of osquery are you using?

osquery master (3ced373e1137fd2403d5cea5ef7d61f4f4fa1248)

### What steps did you take to reproduce the issue?

1. Be sure to stop auditd with `sudo systemctl stop auditd`
2. Run `sudo auditctl -D` to be extra sure there aren't any rules applied
3. Run osquery configured for auditd (shell is fine), for instance: `sudo osquery/osqueryd --disable_events=false --disable_audit=false --audit_allow_process_events=true --audit_allow_config=true --audit_allow_fork_process_events --audit_allow_sockets --verbose`
4. Verify that you can see the `Enabling audit rules for [...]` message, then close osquery either via Ctrl-C or `.q`
5. Check the installed rules with `sudo auditctl -l`

### What did you expect to see?

I expected to see no rules installed

### What did you see instead?

When closing osquery prints:
`I0722 12:18:30.712779 25830 auditdnetlink.cpp:578] Uninstalling the audit rules we have installed`

But then:
```
sudo auditctl -l                                                                                                                                                                                               
-a always,exit -S connect,bind,clone,fork,vfork,execve,execveat
```

Rules are still there.

**Comment by Smjert** [comment]
nvm should've checked better the code, I think there are some descriptions to be updated about some flags, because they are confusing.

**Comment by Smjert** [comment]
Reopening, since it's actually a regression from https://github.com/osquery/osquery/pull/7063.

I've also noticed that before the regression we were also adding rules twice in the list of rules to remove if the `FLAGS_audit_force_unconfigure` flag was active.

**Comment by prateeknischal** [comment]
Hi @Smjert , This seems to be my doing and because of a misunderstanding of the `--audit_force_unconfigure` flag. Really sorry for that. I have attempted to fix it in #7221 .

Please let me know if the fix looks good.
